myapp/validates/UserValidation.js

```diff
@@ -0,0 +1,138 @@
+const USER_ALLOWED_INPUT_FIELDS = new Set(
+  require('config').get('User.USER_ALLOWED_INPUT_FIELDS')
+);
+
+const ADDRESS_ALLOWED_INPUT_FIELDS = new Set(
+  require('config').get('User.ADDRESS_ALLOWED_INPUT_FIELDS')
+);
+
+const VALID_DATE_FORMAT = /^(\d{4})-(\d{2})-(\d{2})$/;
+const VALID_INTEGER = /^\d+$/;
+const VALID_GEO_COORDINATES = /^-?[0-9]{1,3}(?:\.[0-9]{1,4})?$/;
+
+class UserValidation {
+
+  /**
+   * Validate if the input longtitude is between -180 to 180.
+   * @param {String} longtitude 
+   * @returns {Boolean}
+   */
+  static isValidLongtitude(longtitude) {
+    if (VALID_GEO_COORDINATES.test(longtitude)) {
+      const val = parseFloat(longtitude);
+      if (val >= -180.0 && val <= 180.0) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Validate if the input latitude is between -85 to 85.
+   * @param {String} latitude 
+   * @returns {Boolean}
+   */
+  static isValidLatitude(latitude) {
+    if (VALID_GEO_COORDINATES.test(latitude)) {
+      const val = parseFloat(latitude);
+      if (val >= -85.0 && val <= 85.0) {
```

Comment:
Does it mean people located at North/South Pole won't be able to use this application? https://www.esrl.noaa.gov/gmd/grad/antuv/SouthPole.jsp

Reply:
* Since the current version of the application is taking advantages of GEOADD and GEOSERACH commands in redis to input the user's geo coordinates and calculate his nearby friends, there is some limitation for GEOADD command as follows:

> The command takes arguments in the standard format x,y so the longitude must be specified before the latitude. There are limits to the coordinates that can be indexed: areas very near to the poles are not indexable.

> The exact limits, as specified by EPSG:900913 / EPSG:3785 / OSGEO:41001 are the following:

> Valid longitudes are from -180 to 180 degrees.
> Valid latitudes are from -85.05112878 to 85.05112878 degrees.
> The command will report an error when the user attempts to index coordinates outside the specified ranges.  

* So that was the reason why the -85.0 and 85.0 validation for the latitude came in the first place. Unfortunately for the people around north/south poles, they cannot use the current version of this application :)

 ## Options to handle the coordinates near the north or the south pole:

This question does bring a good point on how to index for the coordinates near the north or the south pole. Here are a few options that I could think of:

* We could still use GEOADD/GEOSEARCH from redis to fetch our nearby friends when the latitude of those friends are not very large. According to the precision, the error rate from GEOHASH algorithm is as follows:
  * At the Equator (0 Degrees), the length of a longitude degree is 111.320 km, while a degree of latitude measures 110.574 km, an error of 0.67%.
  * At 30 Degrees (Mid Latitudes) the error is 110.852/96.486 = 14.89%.
  * At 60 Degrees (High Arctic) the error is 111.412/55.800 = 99.67%, reaching infinity at the poles.
* For edge cases like people with larger latitude, we could switch to PostGIS, which is a spatial database extended from postgreSql. It has rich support to spartial projecting and provides excellent functions to solve those edge cases.